integrationTests/vm/arwen/delegation/delegationSimulation_test.go

```diff
@@ -44,214 +24,30 @@ func TestSimulateExecutionOfStakeTransactionAndQueries(t *testing.T) {
 }
 
 func runDelegationExecutionSimulate(t *testing.T, numRuns uint32, numBatches uint32, numTxPerBatch uint32, numQueriesPerBatch uint32) {
-	cacheConfig := storageUnit.CacheConfig{
-		Name:        "trie",
-		Type:        "SizeLRU",
-		SizeInBytes: 314572800, //300MB
-		Capacity:    500000,
-	}
-	trieCache, err := storageUnit.NewCache(cacheConfig)
-	require.Nil(t, err)
-	dbConfig := config.DBConfig{
-		FilePath:          "trie",
-		Type:              "LvlDBSerial",
-		BatchDelaySeconds: 2,
-		MaxBatchSize:      45000,
-		MaxOpenFiles:      10,
-	}
-	persisterFactory := factory.NewPersisterFactory(dbConfig)
-	tempDir, err := ioutil.TempDir("", "integrationTest")
-	require.Nil(t, err)
-	triePersister, err := persisterFactory.Create(tempDir)
-	require.Nil(t, err)
-	trieStorage, err := storageUnit.NewStorageUnit(trieCache, triePersister)
-	require.Nil(t, err)
-
-	defer func() {
-		err = trieStorage.DestroyUnit()
-		log.LogIfError(err)
-	}()
-
-	gasMap, err := core.LoadGasScheduleConfig("../../../../cmd/node/config/gasSchedules/gasScheduleV2.toml")
-	require.Nil(t, err)
-
-	node := integrationTests.NewTestProcessorNodeWithStorageTrieAndGasModel(
-		1,
-		0,
-		0,
-		"",
-		trieStorage,
-		gasMap,
-	)
-
-	totalSupply, _ := big.NewInt(0).SetString("20000000000000000000000000", 10) //20MIL eGLD
-	nodeInitialBalance := big.NewInt(0).Set(totalSupply)
-	nodeInitialBalance.Div(nodeInitialBalance, big.NewInt(2))
-	node.EconomicsData.SetMaxGasLimitPerBlock(1500000000)
-	node.EconomicsData.SetMinGasLimit(50000)
-	node.EconomicsData.SetMinGasPrice(1000000000)
-	node.EconomicsData.SetTotalSupply(totalSupply)
-	integrationTests.MintAllNodes([]*integrationTests.TestProcessorNode{node}, nodeInitialBalance)
-
-	numAccounts := 100000
-	accountsInitialBalance, _ := big.NewInt(0).SetString("1000000000000000000000", 10) //1000eGLD
-	addresses := generateAndMintAccounts(t, node, accountsInitialBalance, numAccounts)
-
-	delegationAddr, err := node.BlockchainHook.NewAddress(node.OwnAccount.Address, node.OwnAccount.Nonce, []byte{5, 0})
-	log.Info("delegation contract", "address", integrationTests.TestAddressPubkeyConverter.Encode(delegationAddr))
-
-	deployDelegationSC(t, node)
-
-	_, err = node.AccntState.Commit()
-	require.Nil(t, err)
-
-	stopRequest := make(chan struct{}, 1)
-	wg := sync.WaitGroup{}
-	if numQueriesPerBatch > 0 {
-		wg.Add(1)
-		copiedAddresses := make([][]byte, 0)
-		for _, address := range addresses {
-			copiedAddresses = append(copiedAddresses, address)
+	gasMapFilename := "../../../../cmd/node/config/gasSchedules/gasScheduleV2.toml"
+	delegationScFilename := "../testdata/delegation/delegation_v0_5_2_full.wasm"
+	benchmarks, err := RunDelegationStressTest(numRuns, numBatches, numTxPerBatch, numQueriesPerBatch, gasMapFilename, delegationScFilename)
```

Comment:
it would be better to give the prepared gasSchedule instead of the file.

Reply:
done, uniformized to resemble the `RunTest` method